packages/firebase_admob/lib/firebase_admob.dart

```diff
@@ -82,6 +82,65 @@ class MobileAdTargetingInfo {
   }
 }
 
+// The types of ad sizes supported for banners.
+enum AdSizeType {
+  WidthAndHeight,
+  SmartBanner,
+}
+
+/// [AdSize] represents the size of a banner ad.
+class AdSize {
+  final int height;
+  final int width;
+  final AdSizeType sizeType;
+
+  /// Creates a typical width-by-height banner size. In most cases, it's easier
+  /// to use one of the named constructors like [AdSize.banner].
+  AdSize({this.width, this.height})
+      : assert(width > 0),
+        assert(height > 0),
```

Comment:
I don't think you want to expose this constructor, since all of the valid ad sizes are covered by the other constructors.

It should be a (private) const constructor where all of the parameters are `@required`.

Reply:
Done. If the plugin is ever expanded to handle DoubleClick for Publishers, it'll need to be made public again, but this is just fine for AdMob.